src/test/java/com/amazonaws/auth/DefaultAWSCredentialsProviderChain.java

```diff
@@ -0,0 +1,23 @@
+package com.amazonaws.auth;
+
+import static java.lang.System.getenv;
+
+/**
+ * Shameless copy of the original for testing {@link io.fabric8.maven.docker.util.aws.AwsSdkAuthConfigFactory}.
+ * Based on <tt>com.amazonaws:aws-java-sdk-core:1.11.707</tt>.
```

Comment:
Are there any licensing issues ? Under which license the aws-java-sdk-core  is licensed (so that it allows code copying with contribution ?). Don't wanna get into struggle because of licensing (was there already :)

Is there any chance that we could implement this without reusing SDK code ?

Reply:
I know its only a test class, but still this could pose some issue for some folks. The good thing is that we are talking here about a tool which is not part the of any delivered application that builds with it.

Reply:
The AWS SDK is licensed under [Apache License 2.0](https://github.com/aws/aws-sdk-java/blob/master/LICENSE.txt). Afaik that's one of the most liberal licenses you could ask for.

I could remove this, but then there wouldn't be a test at all. What do you think?

Reply:
Ok, APL is ok. I had once a case where a GPL code snippet sneaked in (in another project). Was quite some hazzle to get this out again. I'm fine with it now (I will add some extra comment here)

Reply:
Yeah, licensing can be a pain...